src/zabapgit_object_iamu.prog.abap

```diff
@@ -0,0 +1,321 @@
+
+  METHOD lif_object~has_changed_since.
+
+    rv_changed = abap_true.
+
+  ENDMETHOD.
+
+  METHOD lif_object~changed_by.
+
+    rv_user = c_user_unknown.
+
+  ENDMETHOD.
+
+  METHOD lif_object~get_metadata.
+
+    rs_metadata = get_metadata( ).
+
+  ENDMETHOD.
+
+  METHOD lif_object~serialize.
+
+    DATA: ls_internet_appl_comp_binary TYPE ty_internet_appl_comp_binary.
+
+    ls_internet_appl_comp_binary = read( ).
+
+    io_xml->add( iv_name = 'IAMU'
+                 ig_data = ls_internet_appl_comp_binary ).
+
+  ENDMETHOD.
+
+  METHOD lif_object~deserialize.
+
+    DATA: ls_internet_appl_comp_binary TYPE ty_internet_appl_comp_binary.
+
+    io_xml->read(
+      EXPORTING
+        iv_name = 'IAMU'
+      CHANGING
+        cg_data = ls_internet_appl_comp_binary ).
+
+    ls_internet_appl_comp_binary-attributes-devclass = iv_package.
+
+    save( ls_internet_appl_comp_binary ).
+
+  ENDMETHOD.
+
+  METHOD lif_object~delete.
+
+    load_mime_api( ).
+
+    mo_mime_api->if_w3_api_object~set_changeable(
+      EXPORTING
+        p_changeable                 = abap_true
+      EXCEPTIONS
+        action_cancelled             = 1
+        object_locked_by_other_user  = 2
+        permission_failure           = 3
+        object_already_changeable    = 4
+        object_already_unlocked      = 5
+        object_just_created          = 6
+        object_deleted               = 7
+        object_modified              = 8
+        object_not_existing          = 9
+        object_invalid               = 10
+        error_occured                = 11
+        content_data_error           = 12
+        OTHERS                       = 13 ).
+
+    IF sy-subrc <> 0.
+      zcx_abapgit_exception=>raise( |Error from if_w3_api_mime~set_changeable| ).
+    ENDIF.
+
+    mo_mime_api->if_w3_api_object~delete(
+      EXCEPTIONS
+        object_not_empty      = 1
+        object_not_changeable = 2
+        object_invalid        = 3
+        error_occured         = 4
+        OTHERS                = 5 ).
+    IF sy-subrc <> 0.
+      zcx_abapgit_exception=>raise( |Error from if_w3_api_mime~delete| ).
+    ENDIF.
+
+    mo_mime_api->if_w3_api_object~save(
+      EXCEPTIONS
+        object_invalid        = 1
+        object_not_changeable = 2
+        action_cancelled      = 3
+        permission_failure    = 4
+        not_changed           = 5
+        data_invalid          = 6
+        error_occured         = 7
+        OTHERS                = 8 ).
+
+    IF sy-subrc <> 0.
+      zcx_abapgit_exception=>raise( |Error from if_w3_api_mime~save| ).
+    ENDIF.
+
+  ENDMETHOD.
+
+  METHOD lif_object~exists.
+
+    DATA: ls_mime_name TYPE iacikeym.
+
+    ls_mime_name = ms_item-obj_name.
+
+    TRY.
+        cl_w3_api_mime=>s_check_exist(
+          EXPORTING
+            p_mime_name = ls_mime_name
+          IMPORTING
+            p_exists    = rv_bool ).
+
+      CATCH zcx_abapgit_exception.
+        rv_bool = abap_false.
+        RETURN.
```

Comment:
this RETURN is not needed, better just to have the method return normally

Reply:
Can `cl_w3_api_mime=>s_check_exist` even raise `zcx_abapgit_exception`? The try-catch doesn't make sense to me.

Reply:
yeah, really need to finish https://github.com/larshp/abapOpenChecks/issues/109 sometime

Reply:
You're right the try catch block doesn't make any sense. It's an relict from a previous implementation. I'll remove it.